docs/source/complex_numbers.rst

```diff
@@ -0,0 +1,174 @@
+     If the default floating point dtype is torch.float64 then complex numbers are inferred to
+     have a dtype of torch.complex128, otherwise they are assumed to have a dtype of torch.complex64.
+
+All factory functions apart from :func:`torch.linspace`, :func:`torch.logspace`, and :func:`torch.arange` are
+supported for complex tensors.
+
+Transition from the old representation
+--------------------------------------
+
+Users who currently worked around the lack of complex tensors with real tensors of shape `(..., 2)`
+can easily to switch using the complex tensors in their code using :func:`torch.view_as_complex` and
+- :func:`torch.view_as_real`. Note that these functions don't perform any copy and
+return a view of the input Tensor.
+
+::
+
+     >>> x = torch.randn(3, 2)
+     >>> x
+     tensor([[ 0.6125, -0.1681],
+          [-0.3773,  1.3487],
+          [-0.0861, -0.7981]])
+     >>> y = torch.view_as_complex(x)
+     >>> y
+     tensor([ 0.6125-0.1681j, -0.3773+1.3487j, -0.0861-0.7981j])
+     >>> torch.view_as_real(y)
+     tensor([[ 0.6125, -0.1681],
+          [-0.3773,  1.3487],
+          [-0.0861, -0.7981]])
+
+Accessing real and imag
+-----------------------
+
+The real and imaginary values of a complex tensor can be accessed using the :attr:`real` and
+:attr:`imag` views.
+
+::
+
+     >>> y.real
+     tensor([ 0.6125, -0.3773, -0.0861])
+     >>> y.imag
+     tensor([-0.1681,  1.3487, -0.7981])
+
+Angle and abs
+-------------
+
+The angle and absolute values of a complex tensor can be accessed using :func:`torch.angle` and
+`torch.abs`.
+
+::
+
+     >>> x1=torch.tensor([3j, 4+4j])
+     >>> x1.abs()
+     tensor([3.0000, 5.6569])
+     >>> x1.angle()
+     tensor([1.5708, 0.7854])
+
+Linear Algebra
+--------------
+
+Currently, there is very minimal linear algebra operation support for complex tensors.
+We currently support :func:`torch.mv`, :func:`torch.svd`, :func:`torch.qr`, and :func:`torch.inverse`
+(the latter three are only supported on CPU). However we are working to add support for more
+functions soon: :func:`torch.matmul`, :func:`torch.solve`, :func:`torch.eig`, :func:`torch.eig`,
+:func:`torch.symeig`. If any of these would help your use case, please
+`search <https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+complex>`_
+if an issue has already been filed and if not, `file one <https://github.com/pytorch/pytorch/issues/new/choose>`_.
+
+
+Serialization
+-------------
+
+Complex Tensors can be serialized, allowing data to be saved as complex values.
+
+::
+
+     >>> torch.save(y, 'complex_tensor.pt')
+     >>> torch.load('complex_tensor.pt')
+     tensor([ 0.6125-0.1681j, -0.3773+1.3487j, -0.0861-0.7981j])
+
+
+Autograd
+--------
+
+PyTorch supports Autograd for Complex Tensors. The autograd APIs can be
+used for both holomorphic and non-holomorphic functions. For non-holomorphic
+functions, the gradient is evaluated as if it were holomorphic. For more details,
```

Comment:
not sure if it's meaningful to say "evaluated as if it were holomorphic"; what does that even mean?

Reply:
ha I meant to say the gradient evaluation for non-holomorphic functions are done assuming the input function is holomorphic. I'll re-phrase to make it clear